File: problems/1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        sortedArr = sorted(arr.copy())
        rankHash = {}
        rank = 1

        for num in sortedArr:
            if num not in rankHash:
                rankHash[num] = rank
                rank += 1
        
        return [rankHash[num] for num in arr]

File: problems/1256-rank-transform-of-an-array/rank-transform-of-an-array.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
const arrayRankTransform = arr => {
    const sortArr = [...arr].sort((a, b) => a - b);
    const rankMap = new Map();
    let rank = 1;
    
    for (let num of sortArr) {
        if (!rankMap.has(num)) {
            rankMap.set(num, rank++);
        }
    }
    
    for (let i = 0; i < arr.length; i++) {
        arr[i] = rankMap.get(arr[i]);
    }
    
    return arr;
};